enterprise_gateway/services/processproxies/processproxy.py

```diff
@@ -586,18 +587,35 @@ def _get_ssh_client(self, host):
         global remote_user
         global remote_pwd
         if remote_user is None:
-            remote_user = os.getenv('EG_REMOTE_USER', getpass.getuser())
-            remote_pwd = os.getenv('EG_REMOTE_PWD')  # this should use password-less ssh
+            use_gss = os.getenv("EG_REMOTE_GSS_SSH", None)
+            remote_pwd = os.getenv("EG_REMOTE_PWD")  # this should use password-less ssh
+            remote_user = os.getenv("EG_REMOTE_USER", getpass.getuser())
+
+            if use_gss and (remote_pwd or remote_user):
+                warnings.warn(
+                    "Both `EG_REMOTE_GSS_SSH` and one of `EG_REMOTE_PWD` or `EG_REMOTE_USER` is set. "
+                    "Those options are mutually exclusive, you configuration may be incorrect. "
+                    "EG_REMOTE_GSS_SSH will take priority."
+                )
 
         try:
             ssh = paramiko.SSHClient()
             ssh.load_system_host_keys()
-            ssh.set_missing_host_key_policy(paramiko.RejectPolicy())
             host_ip = gethostbyname(host)
-            if remote_pwd:
-                ssh.connect(host_ip, port=ssh_port, username=remote_user, password=remote_pwd)
+            if use_gss:
+                ssh.set_missing_host_key_policy(paramiko.AutoAddPolicy())
```

Comment:
Should we be automatically accepting the host keys? Is this a requirement of using GSS?

Reply:
I can make it a flag. This is what was requested in #946, and it ismy understanding that with GSS ; kerberos do a custom key exchange that act as authentication of both system.

Reply:
Thanks @Carreau. Since Kerberos is doing the auth, I am ok with having the auto-add policy.